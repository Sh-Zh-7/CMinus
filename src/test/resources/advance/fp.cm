int main()
{
    float x = 1.1;
    float y = 1.;
    float z = .1;

    float a = 1.2e3;
    float b = .2e-3;
    float c = 1.e+3;
    float d = 1.e03;
    float e = 1.E03;
}