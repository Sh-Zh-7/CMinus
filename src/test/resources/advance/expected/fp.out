Program(1)
  ExtDefList(1)
    ExtDef(1)
      Specifier(1)
        TYPE: int
      FunDec(1)
        ID: main
        LP
        RP
      CompSt(2)
        LC
        DefList(3)
          Def(3)
            Specifier(3)
              TYPE: float
            DecList(3)
              Dec(3)
                VarDec(3)
                  ID: x
                ASSIGNOP
                Exp(3)
                  FLOAT: 1.1
            SEMI
          DefList(4)
            Def(4)
              Specifier(4)
                TYPE: float
              DecList(4)
                Dec(4)
                  VarDec(4)
                    ID: y
                  ASSIGNOP
                  Exp(4)
                    FLOAT: 1.0
              SEMI
            DefList(5)
              Def(5)
                Specifier(5)
                  TYPE: float
                DecList(5)
                  Dec(5)
                    VarDec(5)
                      ID: z
                    ASSIGNOP
                    Exp(5)
                      FLOAT: 0.1
                SEMI
              DefList(7)
                Def(7)
                  Specifier(7)
                    TYPE: float
                  DecList(7)
                    Dec(7)
                      VarDec(7)
                        ID: a
                      ASSIGNOP
                      Exp(7)
                        FLOAT: 1200.0
                  SEMI
                DefList(8)
                  Def(8)
                    Specifier(8)
                      TYPE: float
                    DecList(8)
                      Dec(8)
                        VarDec(8)
                          ID: b
                        ASSIGNOP
                        Exp(8)
                          FLOAT: 2.0E-4
                    SEMI
                  DefList(9)
                    Def(9)
                      Specifier(9)
                        TYPE: float
                      DecList(9)
                        Dec(9)
                          VarDec(9)
                            ID: c
                          ASSIGNOP
                          Exp(9)
                            FLOAT: 1000.0
                      SEMI
                    DefList(10)
                      Def(10)
                        Specifier(10)
                          TYPE: float
                        DecList(10)
                          Dec(10)
                            VarDec(10)
                              ID: d
                            ASSIGNOP
                            Exp(10)
                              FLOAT: 1000.0
                        SEMI
                      DefList(11)
                        Def(11)
                          Specifier(11)
                            TYPE: float
                          DecList(11)
                            Dec(11)
                              VarDec(11)
                                ID: e
                              ASSIGNOP
                              Exp(11)
                                FLOAT: 1000.0
                          SEMI
        RC

